Source/Core/Core/HLE/HLE.cpp

```diff
@@ -235,4 +235,27 @@ u32 UnPatch(std::string_view patch_name)
 
   return 0;
 }
+
+u32 UnpatchRange(u32 start_addr, u32 end_addr)
+{
+  u32 count = 0;
+
+  for (auto i = s_hooked_addresses.begin(); i != s_hooked_addresses.end();)
```

Comment:
`s_hooked_addresses` is a sorted map. Instead of starting at `s_hooked_addresses.begin()` and ending when `i == s_hooked_addresses.end()`, you could start at `s_hooked_addresses.lower_bound(start_addr)` and end when `i == s_hooked_addresses.end() || i->first >= end_addr`.